controllers/hvpa_controller.go

```diff
@@ -280,47 +274,67 @@ func (r *HvpaReconciler) deleteHvpaFinalizers(hvpa *autoscalingv1alpha1.Hvpa) {
 	}
 }
 
-func getVpaFromHvpa(hvpa *autoscalingv1alpha1.Hvpa) (*vpa_api.VerticalPodAutoscaler, error) {
-	metadata := hvpa.Spec.Vpa.Template.ObjectMeta.DeepCopy()
-
-	labels := metadata.GetLabels()
-	if labels == nil || len(labels) == 0 {
-		// TODO: Could be done better as part of validation
-		return nil, fmt.Errorf("Need labels in VPA template")
+func (r *HvpaReconciler) reconcileVpa(hvpa *autoscalingv1alpha1.Hvpa) (*vpa_api.VerticalPodAutoscalerStatus, error) {
+	selector, err := metav1.LabelSelectorAsSelector(hvpa.Spec.Vpa.Selector)
+	if err != nil {
+		log.Error(err, "Error converting vpa selector to selector")
+		return nil, err
 	}
 
-	if ownerRef := metadata.GetOwnerReferences(); len(ownerRef) != 0 {
-		// TODO: Could be done better as part of validation
-		return nil, fmt.Errorf("vpa template in hvpa object already has an owner reference")
+	// list all vpas to include the vpas that don't match the hvpa`s selector
+	// anymore but has the stale controller ref.
+	vpas := &vpa_api.VerticalPodAutoscalerList{}
+	err = r.List(context.TODO(), vpas, client.InNamespace(hvpa.Namespace))
```

Comment:
```suggestion
	listOpts := &client.ListOptions{
	  LabelSelector: selector,
	  Namespace: hvpa.Namespace,
	}
	err = r.List(context.TODO(), vpas, client.InNamespace(hvpa.Namespace), client.UseListOptions(listOpts))
```

With this change selector is not required for filtering later.